File: solutions/elixir/sum-of-multiples/1/lib/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(limit, factors) do
    factors = Enum.reject(factors, &(&1 === 0))

    1..(limit - 1)
    |> Enum.filter(fn item ->
      factors
      |> Enum.map(&rem(item, &1))
      |> Enum.any?(&(&1 === 0))
    end)
    |> Enum.sum()
  end
end

File: solutions/typescript/sum-of-multiples/1/sum-of-multiples.ts
export function sum(multiples: number[], max: number): number {
  let total = 0;
  for ( let i = 1; i < max; i++ ) {
    const matches = multiples.filter((c: number): boolean => {
      return i % c === 0;
    })
    if ( matches.length > 0 ) total += i;
  }
  return total;
}
